test/functional/pdo_sqlsrv/pdo_AE_functions.inc

```diff
@@ -0,0 +1,532 @@
+            if ($encryptionType == 'Deterministic') {
+                if ($comparison == '=') {
+                    print_r($error);
+                    die("Equality comparison failed for deterministic encryption!\n");
+                } else {
+                    $e = $error->errorInfo;
+                    checkErrors($e, array('42000', '33277'));
+                }
+            } elseif (isEnclaveEnabled($key)) {
+                $e = $error->errorInfo;
+                checkErrors($e, array('42000', '33546'));
+            } elseif (!isEnclaveEnabled($key)) {
+                $e = $error->errorInfo;
+                checkErrors($e, array('42000', '33277'));
+            } else {
+                print_r($error);
+                die("AE statement execution when it shouldn't!");
+            }
+        } elseif ($attestation == 'wrongurl') {
+            if ($encryptionType == 'Deterministic') {
+                if ($comparison == '=') {
+                    $e = $error->errorInfo;
+                    die("Equality comparison failed for deterministic encryption!\n");
+                } else {
+                    $e = $error->errorInfo;
+                    checkErrors($e, array('42000', '33277'));
+                }
+            } elseif (isEnclaveEnabled($key)) {
+                $e = $error->errorInfo;
+                checkErrors($e, array('CE405', '0'));
+            } elseif (!isEnclaveEnabled($key)) {
+                $e = $error->errorInfo;
+                checkErrors($e, array('42000', '33277'));
+            } else {
+                print_r($error);
+                die("AE statement execution when it shouldn't!");
+            }
+        } elseif ($attestation == 'correct') {
+            if (!isEnclaveEnabled($key) and $encryptionType == 'Randomized') {
+                $e = $error->errorInfo;
+                checkErrors($e, array('42000', '33277'));
+            } elseif ($encryptionType == 'Deterministic') {
+                if ($comparison == '=') {
+                    print_r($error);
+                    die("Equality comparison failed for deterministic encryption!\n");
+                } else {
+                    $e = $error->errorInfo;
+                    checkErrors($e, array('42000', '33277'));
+                }
+            } else {
+                print_r($error);
+                die("Comparison failed for correct attestation when it shouldn't have!\n");
+            }
+        } else {
+            print_r($error);
+            die("Unexpected error occurred in compareResults!\n");
+        }
+
+        return;
+    }
+
+    $AEres = $AEstmt->fetchAll(PDO::FETCH_NUM);
+    $nonAEres = $nonAEstmt->fetchAll(PDO::FETCH_NUM);
+    $AEcount = count($AEres);
+    $nonAEcount = count($nonAEres);
+
+    if ($type == 'char' or $type == 'nchar') {
+        // char and nchar may not return the same results - at this point
+        // we've verified that statement execution works so just return
```

Comment:
Is the difference something to be fixed in the near future? 
If so, please put a TODO here 
If not, please justify